src/lib.rs

```diff
@@ -1,54 +1,116 @@
-//! This module provides macro `assign!` to allow mutating instance with declarative flavor
+//! This module provides the `assign!` macro to allow mutating a struct value in
+//! a declarative style.
 //!
-//! The motivation of this macro is to enable programmer to document a sequence of mutations instance fields as initialization by writing it in a declarative way. `assign!` macro also allows programmer to skip defining fields that has default value. Such case are used when a dependency is exposing an non-exhaustive struct
+//! It is an alternative to [struct update syntax][] that works with
```

Comment:
I am not sure if it is an alternative to struct update syntax as the prerequisites of both are different although the motivation is similar.
Struct update syntax requires another struct instance for the update,
while assign! simply is a shorthand for updating values.

I think it would be better if framed as:
- assign! supplements struct update syntax (or the other way around?) for struct NOT marked non_exhaustive
- assign! is a workaround

Reply:
I don't understand what you mean. For non-non_exhaustive structs,

```
assign!(init, { fields... });
```

has exactly the same effect as

```
Foo { fields..., ..init }
```

Reply:
Struct update requires an existing struct in exactly the same way that `assign!` requires it, that is you can also write

```
Foo { a: 5, ..Default::default() }
```

And in fact that is invalid the most common user case for it.

Reply:
I was just not sure if it is an `alternative` because the struct update syntax seems to be non-zero-cost as it seems to instantiate a struct twice and it was not to be released soon.
Then I built assign so that it can be a zero-cost way to do that and soon.

My curiousity is answered here, thought. So I think my question is resolved

https://stackoverflow.com/questions/48212841/runtime-cost-of-creating-new-struct-with-update-syntax-in-rust